pkgs/by-name/ec/echidna/package.nix

```
@@ -0,0 +1,124 @@
+{
+  lib,
+  stdenv,
+  makeWrapper,
+  haskellPackages,
+  fetchFromGitHub,
+  # dependencies
+  slither-analyzer,
+}:
+
+haskellPackages.mkDerivation rec {
+  pname = "echidna";
+  version = "2.2.6";
+
+  src = fetchFromGitHub {
+    owner = "crytic";
+    repo = "echidna";
+    tag = "v${version}";
+    sha256 = "sha256-5nzis7MXOqs0bhx2jrEexjZYZI2qY6D0D7AWO+SPs+A=";
+  };
+
+  isExecutable = true;
+
+  buildTools = with haskellPackages; [
+    hpack
+  ];
+
+  executableHaskellDepends = with haskellPackages; [
+    # base dependencies
+    aeson
+    base
+    containers
+    directory
+    hevm
+    MonadRandom
+    mtl
+    text
+    # library dependencies
+    ansi-terminal
+    async
+    base16-bytestring
+    binary
+    brick
+    bytestring
+    data-bword
+    data-dword
+    deepseq
+    exceptions
+    extra
+    filepath
+    hashable
+    html-conduit
+    html-entities
+    http-conduit
+    ListLike
+    optics
+    optics-core
+    process
+    random
+    rosezipper
+    semver
+    signal
+    split
+    strip-ansi-escape
+    time
+    unliftio
+    utf8-string
+    vector
+    vty
+    vty-crossplatform
+    wai-extra
+    warp
+    word-wrap
+    xml-conduit
+    yaml
+    # executable dependencies
+    code-page
+    filepath
+    hashable
+    optparse-applicative
+    time
+    with-utf8
+  ];
+
+  executableToolDepends = [
+    makeWrapper
+  ];
+
+  preConfigure = ''
+    hpack
+  '';
+
+  postInstall =
+    with haskellPackages;
+    # https://github.com/NixOS/nixpkgs/pull/304352
+    lib.optionalString (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64) ''
+      remove-references-to -t ${warp.out} "$out/bin/echidna"
+      remove-references-to -t ${wreq.out} "$out/bin/echidna"
+    ''
+    # make slither-analyzer a runtime dependency
+    + ''
+      wrapProgram $out/bin/echidna \
+        --set PATH ${
```

Usually better instead so e.g. coreutils are still accessible.

```suggestion
        --set PREFIX PATH : ${
```

---

I guess you meant `--prefix`; one sec, let me fix locally then  push

---

```
$ cat result/bin/echidna 
#! /nix/store/4fvc5fm8bszmkydng1ivrvr5cbvr1g60-bash-5.2p37/bin/bash -e
PATH=${PATH:+':'$PATH':'}
PATH=${PATH/':''/nix/store/8p8wm7y1p2sw0005bawddack82x01krb-python3.12-slither-analyzer-0.10.3/bin'':'/':'}
PATH='/nix/store/8p8wm7y1p2sw0005bawddack82x01krb-python3.12-slither-analyzer-0.10.3/bin'$PATH
PATH=${PATH#':'}
PATH=${PATH%':'}
export PATH
exec -a "$0" "/nix/store/b0h9mhpvm1sjvgzk0bisx8i95nfs4j11-echidna-2.2.6/bin/.echidna-wrapped"  "$@" 
```

Done.